### PR TITLE
#79: fix data race for blsagg AVS-44

### LINKED ISSUE #79: Data race
Running this on `master`

```go
go test ./... -race
...
==================
WARNING: DATA RACE
Write at 0x00c00007cde8 by goroutine 17:
  math/big.(*Int).Add()
      /usr/local/Cellar/go/1.21.4/libexec/src/math/big/int.go:147 +0x187
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.(*BlsAggregatorService).singleTaskAggregatorGoroutineFunc()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg.go:261 +0x1c44
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.(*BlsAggregatorService).InitializeNewTask.func1()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg.go:161 +0xf3

Previous read at 0x00c00007cde8 by goroutine 18:
  math/big.(*Int).Mul()
      /usr/local/Cellar/go/1.21.4/libexec/src/math/big/int.go:194 +0x1b5
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.checkIfStakeThresholdsMet()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg.go:370 +0x1ab
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.(*BlsAggregatorService).singleTaskAggregatorGoroutineFunc()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg.go:268 +0x11de
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.(*BlsAggregatorService).InitializeNewTask.func1()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg.go:161 +0xf3

Goroutine 17 (running) created at:
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.(*BlsAggregatorService).InitializeNewTask()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg.go:161 +0x2e6
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.TestBlsAgg.func4()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg_test.go:186 +0x805
  testing.tRunner()
      /usr/local/Cellar/go/1.21.4/libexec/src/testing/testing.go:1595 +0x238
  testing.(*T).Run.func1()
      /usr/local/Cellar/go/1.21.4/libexec/src/testing/testing.go:1648 +0x44

Goroutine 18 (running) created at:
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.(*BlsAggregatorService).InitializeNewTask()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg.go:161 +0x2e6
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.TestBlsAgg.func4()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg_test.go:190 +0x8b4
  testing.tRunner()
      /usr/local/Cellar/go/1.21.4/libexec/src/testing/testing.go:1595 +0x238
  testing.(*T).Run.func1()
      /usr/local/Cellar/go/1.21.4/libexec/src/testing/testing.go:1648 +0x44
==================
==================
WARNING: DATA RACE
Write at 0x00c00007cde0 by goroutine 17:
  math/big.(*Int).Add()
      /usr/local/Cellar/go/1.21.4/libexec/src/math/big/int.go:158 +0x4ed
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.(*BlsAggregatorService).singleTaskAggregatorGoroutineFunc()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg.go:261 +0x1c44
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.(*BlsAggregatorService).InitializeNewTask.func1()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg.go:161 +0xf3

Previous read at 0x00c00007cde0 by goroutine 18:
  math/big.(*Int).Mul()
      /usr/local/Cellar/go/1.21.4/libexec/src/math/big/int.go:195 +0x2b4
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.checkIfStakeThresholdsMet()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg.go:370 +0x1ab
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.(*BlsAggregatorService).singleTaskAggregatorGoroutineFunc()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg.go:268 +0x11de
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.(*BlsAggregatorService).InitializeNewTask.func1()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg.go:161 +0xf3

Goroutine 17 (running) created at:
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.(*BlsAggregatorService).InitializeNewTask()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg.go:161 +0x2e6
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.TestBlsAgg.func4()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg_test.go:186 +0x805
  testing.tRunner()
      /usr/local/Cellar/go/1.21.4/libexec/src/testing/testing.go:1595 +0x238
  testing.(*T).Run.func1()
      /usr/local/Cellar/go/1.21.4/libexec/src/testing/testing.go:1648 +0x44

Goroutine 18 (running) created at:
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.(*BlsAggregatorService).InitializeNewTask()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg.go:161 +0x2e6
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.TestBlsAgg.func4()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg_test.go:190 +0x8b4
  testing.tRunner()
      /usr/local/Cellar/go/1.21.4/libexec/src/testing/testing.go:1595 +0x238
  testing.(*T).Run.func1()
      /usr/local/Cellar/go/1.21.4/libexec/src/testing/testing.go:1648 +0x44
==================
==================
WARNING: DATA RACE
Read at 0x00c00007cdc0 by goroutine 18:
  math/big.(*Int).Add()
      /usr/local/Cellar/go/1.21.4/libexec/src/math/big/int.go:143 +0x4d
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.(*BlsAggregatorService).singleTaskAggregatorGoroutineFunc()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg.go:261 +0x1c44
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.(*BlsAggregatorService).InitializeNewTask.func1()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg.go:161 +0xf3

Previous write at 0x00c00007cdc0 by goroutine 17:
  math/big.(*Int).Add()
      /usr/local/Cellar/go/1.21.4/libexec/src/math/big/int.go:158 +0x4ed
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.(*BlsAggregatorService).singleTaskAggregatorGoroutineFunc()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg.go:261 +0x1c44
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.(*BlsAggregatorService).InitializeNewTask.func1()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg.go:161 +0xf3

Goroutine 18 (running) created at:
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.(*BlsAggregatorService).InitializeNewTask()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg.go:161 +0x2e6
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.TestBlsAgg.func4()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg_test.go:190 +0x8b4
  testing.tRunner()
      /usr/local/Cellar/go/1.21.4/libexec/src/testing/testing.go:1595 +0x238
  testing.(*T).Run.func1()
      /usr/local/Cellar/go/1.21.4/libexec/src/testing/testing.go:1648 +0x44

Goroutine 17 (running) created at:
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.(*BlsAggregatorService).InitializeNewTask()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg.go:161 +0x2e6
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.TestBlsAgg.func4()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg_test.go:186 +0x805
  testing.tRunner()
      /usr/local/Cellar/go/1.21.4/libexec/src/testing/testing.go:1595 +0x238
  testing.(*T).Run.func1()
      /usr/local/Cellar/go/1.21.4/libexec/src/testing/testing.go:1648 +0x44
==================
==================
WARNING: DATA RACE
Read at 0x00c00007cdc8 by goroutine 18:
  math/big.(*Int).Add()
      /usr/local/Cellar/go/1.21.4/libexec/src/math/big/int.go:147 +0xb7
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.(*BlsAggregatorService).singleTaskAggregatorGoroutineFunc()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg.go:261 +0x1c44
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.(*BlsAggregatorService).InitializeNewTask.func1()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg.go:161 +0xf3

Previous write at 0x00c00007cdc8 by goroutine 17:
  math/big.(*Int).Add()
      /usr/local/Cellar/go/1.21.4/libexec/src/math/big/int.go:147 +0x187
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.(*BlsAggregatorService).singleTaskAggregatorGoroutineFunc()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg.go:261 +0x1c44
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.(*BlsAggregatorService).InitializeNewTask.func1()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg.go:161 +0xf3

Goroutine 18 (running) created at:
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.(*BlsAggregatorService).InitializeNewTask()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg.go:161 +0x2e6
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.TestBlsAgg.func4()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg_test.go:190 +0x8b4
  testing.tRunner()
      /usr/local/Cellar/go/1.21.4/libexec/src/testing/testing.go:1595 +0x238
  testing.(*T).Run.func1()
      /usr/local/Cellar/go/1.21.4/libexec/src/testing/testing.go:1648 +0x44

Goroutine 17 (running) created at:
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.(*BlsAggregatorService).InitializeNewTask()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg.go:161 +0x2e6
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.TestBlsAgg.func4()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg_test.go:186 +0x805
  testing.tRunner()
      /usr/local/Cellar/go/1.21.4/libexec/src/testing/testing.go:1595 +0x238
  testing.(*T).Run.func1()
      /usr/local/Cellar/go/1.21.4/libexec/src/testing/testing.go:1648 +0x44
==================
==================
WARNING: DATA RACE
Write at 0x00c00002f2c8 by goroutine 18:
  math/big.nat.add()
      /usr/local/Cellar/go/1.21.4/libexec/src/math/big/nat.go:123 +0x264
  math/big.(*Int).Add()
      /usr/local/Cellar/go/1.21.4/libexec/src/math/big/int.go:147 +0x168
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.(*BlsAggregatorService).singleTaskAggregatorGoroutineFunc()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg.go:261 +0x1c44
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.(*BlsAggregatorService).InitializeNewTask.func1()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg.go:161 +0xf3

Previous write at 0x00c00002f2c8 by goroutine 17:
  math/big.nat.add()
      /usr/local/Cellar/go/1.21.4/libexec/src/math/big/nat.go:123 +0x264
  math/big.(*Int).Add()
      /usr/local/Cellar/go/1.21.4/libexec/src/math/big/int.go:147 +0x168
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.(*BlsAggregatorService).singleTaskAggregatorGoroutineFunc()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg.go:261 +0x1c44
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.(*BlsAggregatorService).InitializeNewTask.func1()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg.go:161 +0xf3

Goroutine 18 (running) created at:
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.(*BlsAggregatorService).InitializeNewTask()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg.go:161 +0x2e6
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.TestBlsAgg.func4()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg_test.go:190 +0x8b4
  testing.tRunner()
      /usr/local/Cellar/go/1.21.4/libexec/src/testing/testing.go:1595 +0x238
  testing.(*T).Run.func1()
      /usr/local/Cellar/go/1.21.4/libexec/src/testing/testing.go:1648 +0x44

Goroutine 17 (running) created at:
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.(*BlsAggregatorService).InitializeNewTask()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg.go:161 +0x2e6
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.TestBlsAgg.func4()
      /Users/estensen/Developer/eigensdk-go/services/bls_aggregation/blsagg_test.go:186 +0x805
  testing.tRunner()
      /usr/local/Cellar/go/1.21.4/libexec/src/testing/testing.go:1595 +0x238
  testing.(*T).Run.func1()
      /usr/local/Cellar/go/1.21.4/libexec/src/testing/testing.go:1648 +0x44
==================
--- FAIL: TestBlsAgg (5.06s)
    --- FAIL: TestBlsAgg/2_concurrent_tasks_2_quorums_2_operators_2_correct_signatures (0.02s)
        testing.go:1465: race detected during execution of test
    testing.go:1465: race detected during execution of test
FAIL
FAIL	github.com/Layr-Labs/eigensdk-go/services/bls_aggregation	7.036s
ok  	github.com/Layr-Labs/eigensdk-go/services/pubkeycompendium	1.877s
ok  	github.com/Layr-Labs/eigensdk-go/signer	17.653s
ok  	github.com/Layr-Labs/eigensdk-go/types	1.598s
FAIL
```

### PR DESCRIPTION
**Pull request by chzyer**
Fixes #79.

### What Changed?

```
==================
WARNING: DATA RACE
Write at 0x00c000189980 by goroutine 17:
  runtime.mapassign()
      /usr/local/go/src/runtime/map.go:579 +0x0
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.(*BlsAggregatorService).singleTaskAggregatorGoroutineFunc()
      /Users/cheney/ata/eigensdk-go/services/bls_aggregation/blsagg.go:307 +0x2139
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.(*BlsAggregatorService).InitializeNewTask.gowrap2()
      /Users/cheney/ata/eigensdk-go/services/bls_aggregation/blsagg.go:184 +0xf3

Previous read at 0x00c000189980 by goroutine 18:
  runtime.mapaccess2()
      /usr/local/go/src/runtime/map.go:457 +0x0
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.checkIfStakeThresholdsMet()
      /Users/cheney/ata/eigensdk-go/services/bls_aggregation/blsagg.go:440 +0x116
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.(*BlsAggregatorService).singleTaskAggregatorGoroutineFunc()
      /Users/cheney/ata/eigensdk-go/services/bls_aggregation/blsagg.go:314 +0x1464
  github.com/Layr-Labs/eigensdk-go/services/bls_aggregation.(*BlsAggregatorService).InitializeNewTask.gowrap2()
      /Users/cheney/ata/eigensdk-go/services/bls_aggregation/blsagg.go:184 +0xf3
```

This data race is caused by reading and writing the same map `operatorsAvsStateDict[].StakePerQuorum`, which is returned by `avsRegistryService.GetOperatorsAvsStateAtBlock` in multiple goroutines.  

In the test case, the implementation of `avsRegistryService` is `FakeAvsRegistryService`, and it always returns the same map for [`GetOperatorsAvsStateAtBlock`](https://github.com/Layr-Labs/eigensdk-go/blob/1ac246a1b715cf2d0ee5fc23612f4cacd7248db8/services/avsregistry/avsregistry_fake.go#L41).

We can fix this by cloning a new map for [`StakePerQuorum`](https://github.com/Layr-Labs/eigensdk-go/blob/1ac246a1b715cf2d0ee5fc23612f4cacd7248db8/services/bls_aggregation/blsagg.go#L292).

In the production environment, we can’t ensure that `avsRegistryService.GetOperatorsAvsStateAtBlock` always returns a new map that is safe to modify since `avsRegistryService` is an interface. For example, we have implemented a cache wrapper for `avsRegistryService`.

In conclusion, I think it’s worth cloning the `StakePerQuorum` map for safety since the overhead is low.

### Reviewer Checklist
- [ ] Code is well-documented
- [ ] Code adheres to Go [naming conventions](https://go.dev/doc/effective_go#names)
- [ ] Code deprecates any old functionality before removing it

### CODE REVIEW

**Review by samlaf — APPROVED** [review]
Beautiful, thank you so much.
Another instance where rust's borrow checker would have prevent this.
I banged my head around this for a while, and didn't think that the fake implementation could be the culprit. Thanks so much!
Just need to rebase and then we can merge

**Comment by chzyer** [comment]
Thanks @samlaf, rebased!
